src/PowerShellEditorServices/Services/Symbols/Visitors/SymbolVisitor.cs

```diff
@@ -0,0 +1,245 @@
+            variableExpressionAst.Extent, // TODO: Maybe parent?
+            _file,
+            isDeclaration: variableExpressionAst.Parent is AssignmentStatementAst or ParameterAst));
+    }
+
+    public override AstVisitAction VisitTypeDefinition(TypeDefinitionAst typeDefinitionAst)
+    {
+        SymbolType symbolType = typeDefinitionAst.IsEnum
+            ? SymbolType.Enum
+            : SymbolType.Class;
+
+        IScriptExtent nameExtent = VisitorUtils.GetNameExtent(typeDefinitionAst);
+        return _action(new SymbolReference(
+            symbolType,
+            "type " + typeDefinitionAst.Name,
+            (symbolType is SymbolType.Enum ? "enum " : "class ") + typeDefinitionAst.Name + " { }",
+            nameExtent,
+            typeDefinitionAst.Extent,
+            _file,
+            isDeclaration: true));
+    }
+
+    public override AstVisitAction VisitTypeExpression(TypeExpressionAst typeExpressionAst)
+    {
+        return _action(new SymbolReference(
+            SymbolType.Type,
+            "type " + typeExpressionAst.TypeName.Name,
+            "(type) " + typeExpressionAst.TypeName.Name,
+            typeExpressionAst.Extent,
+            typeExpressionAst.Extent,
+            _file,
+            isDeclaration: false));
+    }
+
+    public override AstVisitAction VisitTypeConstraint(TypeConstraintAst typeConstraintAst)
+    {
+        return _action(new SymbolReference(
+            SymbolType.Type,
+            "type " + typeConstraintAst.TypeName.Name,
+            "(type) " + typeConstraintAst.TypeName.Name,
+            typeConstraintAst.Extent,
+            typeConstraintAst.Extent,
+            _file,
+            isDeclaration: false));
+    }
+
+    public override AstVisitAction VisitFunctionMember(FunctionMemberAst functionMemberAst)
+    {
+        SymbolType symbolType = functionMemberAst.IsConstructor
+            ? SymbolType.Constructor
+            : SymbolType.Method;
+
+        IScriptExtent nameExtent = VisitorUtils.GetNameExtent(functionMemberAst);
+
+        return _action(new SymbolReference(
+            symbolType,
+            "fn " + functionMemberAst.Name, // We bucket all the overloads.
+            VisitorUtils.GetMemberOverloadName(functionMemberAst),
+            nameExtent,
+            functionMemberAst.Extent,
+            _file,
+            isDeclaration: true));
+    }
+
+    public override AstVisitAction VisitPropertyMember(PropertyMemberAst propertyMemberAst)
+    {
+        // Enum members and properties are the "same" according to PowerShell, so the symbol name's
+        // must be the same since we can't distinguish them in VisitMemberExpression.
+        SymbolType symbolType =
+            propertyMemberAst.Parent is TypeDefinitionAst typeAst && typeAst.IsEnum
+                ? SymbolType.EnumMember
+                : SymbolType.Property;
+
+        return _action(new SymbolReference(
+            symbolType,
+            "var " + propertyMemberAst.Name,
```

Comment:
```suggestion
            "prop " + propertyMemberAst.Name,
```

Avoid conflict between properties and variables.